byzcoin/viewchange/viewchange.go

```diff
@@ -167,21 +167,25 @@ func (c *Controller) Start(myID network.ServerIdentityID, genesis skipchain.Skip
 				// To avoid starting the next view-change too
 				// soon, start view-change timer after
 				// receiving 2*f+1 view-change messages.
-				// Watch out for overflow: When ctr gets to be too high, math.Pow(2, float64(str))
+				// Watch out for overflow: When ctr gets to be too high,
+				// math.Pow(2, float64(ctr))
 				// will be too high, and timeout will end up negative.
-				timeout := time.Duration(math.Pow(2, float64(ctr))) * initialDuration
-				if timeout < 0 || timeout.Seconds() > maxTimeout.Seconds() {
-					timeout = maxTimeout
+				timeout := maxTimeout
+				backoff := float64(ctr - 2*f)
```

Comment:
Can you explain that ?

Reply:
Added comment

Reply:
I actually agreed with the math but my question was about the `float64(ctr -2*f)`

Reply:
this code-path supposes that `ctr > 2*f`. So that it doesn't start with `2**(2*f)*timeout`, which can get high very fast, but that it starts with `2**(1)`.

Reply:
Shoot - mixed up `meta.countOf(ctr)` and `ctr` itself...